File: Examples/Color/ViewController.m
//
//  ViewController.m
//  Color
//
//  Created by Burhanuddin Sunelwala on 11/18/14.
//  Copyright (c) 2014 TFT. All rights reserved.
//

#import "ViewController.h"
#import "UIColor+TFT.h"

@interface ViewController () {
    
    __weak IBOutlet UIView *viewRGBHex;
    __weak IBOutlet UIView *viewCMYKHex;
    __weak IBOutlet UIView *viewCMYK;
    
    CGFloat cyan;
    CGFloat magenta;
    CGFloat yellow;
    CGFloat black;
}

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}

- (IBAction)rgbHexTextChanged:(UITextField *)sender {
    
    [viewRGBHex setBackgroundColor:[UIColor colorWithRGBHex:sender.text]];
    NSLog(@"%@", [UIColor hexStringForColor:viewRGBHex.backgroundColor]);
}

- (IBAction)cmykHexTextChanged:(UITextField *)sender {
    
    [viewCMYKHex setBackgroundColor:[UIColor colorWithCMYKHex:sender.text]];
}

- (IBAction)cyanTextChanged:(UITextField *)sender {
    
    cyan = [sender.text floatValue];
    [self updateCMYK];
}
- (IBAction)magentaTextChanged:(UITextField *)sender {
    
    magenta = [sender.text floatValue];
    [self updateCMYK];
}
- (IBAction)yellowTextChanged:(UITextField *)sender {
    
    yellow = [sender.text floatValue];
    [self updateCMYK];
}
- (IBAction)blackTextChanged:(UITextField *)sender {
    
    black = [sender.text floatValue];
    [self updateCMYK];
}

- (void)updateCMYK {
    
    [viewCMYK setBackgroundColor:[UIColor colorWithCyan:cyan/255.0 magenta:magenta/255.0f yellow:yellow/255.0f black:black/255.0f alpha:1.0f]];
}

@end

File: Example/TFTColor/ViewController.swift
//
//  ViewController.swift
//  Swift Example
//
//  Created by Burhanuddin Sunelwala on 12/9/15.
//  Copyright © 2015 Burhanuddin Sunelwala. All rights reserved.
//

import UIKit
import TFTColor

class ViewController: UIViewController {

    @IBOutlet weak var rgbHexTextField: UITextField!
    @IBOutlet weak var cmykHexTextField: UITextField!
    @IBOutlet weak var redTextField: UITextField!
    @IBOutlet weak var greenTextField: UITextField!
    @IBOutlet weak var blueTextField: UITextField!
    @IBOutlet weak var cyanTextField: UITextField!
    @IBOutlet weak var magentaTextField: UITextField!
    @IBOutlet weak var yellowTextField: UITextField!
    @IBOutlet weak var blackTextField: UITextField!
    @IBOutlet weak var colorView: UIView!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        rgbHexTextField.text = "#00ff00"
    
        colorView.backgroundColor = UIColor(rgbHexString: rgbHexTextField.text!, alpha: 1.0)
        let viewColor = colorView.backgroundColor!
        
        cmykHexTextField.text = viewColor.cmykHexString
        
        redTextField.text = "\(viewColor.redValue)"
        greenTextField.text = "\(viewColor.greenValue)"
        blueTextField.text = "\(viewColor.blueValue)"
        
        cyanTextField.text = "\(viewColor.cyanValue)"
        magentaTextField.text = "\(viewColor.magentaValue)"
        yellowTextField.text = "\(viewColor.yellowValue)"
        blackTextField.text = "\(viewColor.blackValue)"
        
        let aview = UIView(frame: CGRect(x: 15, y: 100, width: 100, height: 100))
        aview.backgroundColor = viewColor.complementary
        view.addSubview(aview)
    }
    
    @IBAction func didChange(_ textField: UITextField) {
        
        if let text = textField.text {
            
            switch textField {
                
            case rgbHexTextField:
                colorView.backgroundColor = UIColor(rgbHexString: text, alpha: 1.0)
                let viewColor = colorView.backgroundColor!
                cmykHexTextField.text = viewColor.cmykHexString
                
                redTextField.text = "\(viewColor.redValue)"
                greenTextField.text = "\(viewColor.greenValue)"
                blueTextField.text = "\(viewColor.blueValue)"
                
                cyanTextField.text = "\(viewColor.cyanValue)"
                magentaTextField.text = "\(viewColor.magentaValue)"
                yellowTextField.text = "\(viewColor.yellowValue)"
                blackTextField.text = "\(viewColor.blackValue)"
                
            case cmykHexTextField:
                colorView.backgroundColor = UIColor(cmykHexString: text, alpha: 1.0)
                let viewColor = colorView.backgroundColor!
                rgbHexTextField.text = viewColor.rgbHexString
                
                redTextField.text = "\(viewColor.redValue)"
                greenTextField.text = "\(viewColor.greenValue)"
                blueTextField.text = "\(viewColor.blueValue)"
                
                cyanTextField.text = "\(viewColor.cyanValue)"
                magentaTextField.text = "\(viewColor.magentaValue)"
                yellowTextField.text = "\(viewColor.yellowValue)"
                blackTextField.text = "\(viewColor.blackValue)"
                
            case redTextField, greenTextField, blueTextField:
                colorView.backgroundColor = UIColor(red: CGFloat(Float(redTextField.text!) ?? 0)/255.0,
                                                    green: CGFloat(Float(greenTextField.text!) ?? 0)/255.0,
                                                    blue: CGFloat(Float(blueTextField.text!) ?? 0)/255.0,
                                                    alpha: 1.0)
                let viewColor = colorView.backgroundColor!
                cmykHexTextField.text = viewColor.cmykHexString
                rgbHexTextField.text = viewColor.rgbHexString
                
                cyanTextField.text = "\(viewColor.cyanValue)"
                magentaTextField.text = "\(viewColor.magentaValue)"
                yellowTextField.text = "\(viewColor.yellowValue)"
                blackTextField.text = "\(viewColor.blackValue)"
                
            case cyanTextField, magentaTextField, yellowTextField, blackTextField:
                colorView.backgroundColor = UIColor(cyan: Float(cyanTextField.text!) ?? 0/255.0,
                                                    magenta: Float(magentaTextField.text!) ?? 0/255.0,
                                                    yellow: Float(yellowTextField.text!) ?? 0/255.0,
                                                    black: Float(blackTextField.text!) ?? 0/255.0,
                                                    alpha: 1.0)
                let viewColor = colorView.backgroundColor!
                cmykHexTextField.text = viewColor.cmykHexString
                rgbHexTextField.text = viewColor.rgbHexString
                
                redTextField.text = "\(viewColor.redValue)"
                greenTextField.text = "\(viewColor.greenValue)"
                blueTextField.text = "\(viewColor.blueValue)"
                
            default:
                colorView.backgroundColor = UIColor.white
            }
        }
    }
}




